### PR TITLE
PixelCNN and Codebook Sampling

### PR DESCRIPTION
**Pull request by karalleyna**
## Description

Adds PixelCNN model, PixelCNN training and codebook sampling.

### Figures

I will add the figures once the training is finished.

### Screenshots

Screen Shot 2021-08-12 at 13 39 48

## Steps

- [x] Refactor `vq_vae.py`
- [x] Adds  `run_pixel.py`
- [x] Extends configuration
- [x] Implements  PixelCNNModule in `experiment.py`
- [x] Implements `pixel_cnn.py`

## Checklist:

- [x] Performed a self-review of the code
- [x] Tested on Google Colab.

## Issue

Closes [Issue 352](https://github.com/probml/hermes/issues/352) on Hermes

## Potential problems/Important remarks

- I can finetune the hyperparameters, so they can be changed in the future.

### CODE REVIEW

**Comment by Neoanarika** [comment]
On current inspection, it looks good to me as I pointed out in chat maybe one way to improve sample quality is to use gelu, swish or leaky relu to avoid sparse gradients.
